bridge/ffi/src/c_api.rs

```diff
@@ -0,0 +1,188 @@
+use crate::{bindings::*, utils::*};
+use dsnp_graph_config::SchemaId;
+use dsnp_graph_core::{
+	api::api::{GraphAPI, GraphState},
+	dsnp::{api_types::Action, dsnp_types::DsnpUserId},
+};
+
+#[no_mangle]
+pub extern "C" fn print_hello_graph() {
+	println!("Hello, Graph!");
+}
+
+// Singleton for GraphState
+#[no_mangle]
+static mut GRAPH_STATE: Option<GraphState> = None;
+
+// Intialize GraphState
+#[no_mangle]
+pub unsafe extern "C" fn initialize_graph_state(environment: *const Environment) -> bool {
+	let environment = &*environment;
+	let rust_environment = environment_from_ffi(environment);
+	let graph_state = GraphState::new(rust_environment);
+	GRAPH_STATE = Some(graph_state);
+	true
+}
+
+// Intialize GraphState with capacity
+#[no_mangle]
+pub unsafe extern "C" fn initialize_graph_state_with_capacity(
+	environment: *const Environment,
+	capacity: usize,
+) -> bool {
+	let environment = &*environment;
+	let rust_environment = environment_from_ffi(environment);
+	let graph_state = GraphState::with_capacity(rust_environment, capacity);
+	GRAPH_STATE = Some(graph_state);
+	true
+}
+
+// Get Capacity
+#[no_mangle]
+pub unsafe extern "C" fn get_graph_capacity() -> usize {
+	if let Some(graph_state) = GRAPH_STATE.as_mut() {
+		graph_state.capacity()
+	} else {
+		0
+	}
+}
+
+// State contains user graph
+#[no_mangle]
+pub unsafe extern "C" fn graph_contains_user(user_id: *const DsnpUserId) -> bool {
+	if let Some(graph_state) = GRAPH_STATE.as_mut() {
+		let user_id = &*user_id;
+		graph_state.contains_user_graph(user_id)
+	} else {
+		false
+	}
+}
+
+// Count of users in current graph
+#[no_mangle]
+pub unsafe extern "C" fn graph_users_count() -> usize {
+	if let Some(graph_state) = GRAPH_STATE.as_mut() {
+		graph_state.len()
+	} else {
+		0
+	}
+}
+
+// Remove user
+#[no_mangle]
+pub unsafe extern "C" fn graph_remove_user(user_id: *const DsnpUserId) -> bool {
+	if let Some(graph_state) = GRAPH_STATE.as_mut() {
+		let user_id = &*user_id;
+		graph_state.remove_user_graph(user_id);
+		true
+	} else {
+		false
+	}
+}
+
+//Graph import users data
+#[no_mangle]
+pub unsafe extern "C" fn graph_import_users_data(
+	payloads: *const ImportBundle,
+	payloads_len: usize,
+) -> bool {
+	if let Some(graph_state) = GRAPH_STATE.as_mut() {
+		let payloads = std::slice::from_raw_parts(payloads, payloads_len);
+		let payloads = payloads_from_ffi(&payloads);
+		graph_state.import_users_data(&payloads).is_ok()
```

Comment:
I think this might be related to Error cleanup ticket #28  also since the errors that we expose might need to be mapped to an integer and having enum errors (with an int value) might help with that.